CHANGELOG.md

```diff
@@ -27,12 +27,16 @@ Changelog
 -   Support non-gate instructions (e.g. `MEASURE`) in `to_latex()` (@notmgsk, gh-975).
 -   Test suite has been updated to reduce the use of deprecated features
     (@kilimanjaro, gh-998).
+-   Type hints have been added to `pyquil.gates` and `pyquil.quilatom` modules (@appleby, gh-999).
 
 ### Bugfixes
 
--   Strength two symmetrization was not correctly producing orthogonal 
-    arrays due to erroneous truncation, which has been fixed 
-    (@kylegulshen, gh-990).
+-   Strength two symmetrization was not correctly producing orthogonal arrays due to erroneous
+    truncation, which has been fixed (@kylegulshen, gh-990).
+-   The `STORE` instruction now accepts `int` or `float` in addition to `MemoryReference` as it's
```

Comment:
can you give some context for this and the following entry in the changelog?

Reply:
Added a little something. Lmk what you think.

Reply:
In addition to the changelog -- could you just explain here why these changes are part of this PR? Guessing it's because the type annotations were wrong before w.r.t. the Quil spec? I'm just trying to understand

Reply:
Ah. Yes, these changes were to bring pyquil in line with what the spec says, sorry.

If you "show resolved" on my comments above, I left a little blurb about them. Maybe I should have waited to mark them resolved to leave an e-paper trail :)

Reply:
Hmm, maybe I should check to make sure the rest of the stack supports this. The tests I added really just check that it parses and prints correctly...

Reply:
It looks like quilc also doesn't support `int` or `float` as the `source` argument for `STORE` instructions. Perhaps the bug should be fixed there first before tackling #815 here. I will file a quilc issue and revert the related changes in the this diff, leaving a comment in the source.

The change to allow floats as the third argument of the comparison operators seem to be OK. Quilc handles them correctly, and I can pass them through `ForestConnection._qvm_run` and get back reasonable results.

Reply:
Filed: https://github.com/rigetti/quilc/issues/425

Reply:
>  I will file a quilc issue and revert the related changes in the this diff, leaving a comment in the source.

I have a pending PR that fixes the bug in quilc, so I'm leaving the fix in place here as well, rather than reverting.

Reply:
The [corresponding quilc PR](https://github.com/rigetti/quilc/pull/428) has now been merged.